community/kernel/src/test/java/org/neo4j/kernel/impl/api/store/StorePropertyCursorTest.java

```diff
@@ -0,0 +1,493 @@
+import org.neo4j.graphdb.mockfs.EphemeralFileSystemAbstraction;
+import org.neo4j.io.pagecache.PageCache;
+import org.neo4j.io.pagecache.PageCursor;
+import org.neo4j.kernel.api.cursor.PropertyItem;
+import org.neo4j.kernel.impl.store.DynamicArrayStore;
+import org.neo4j.kernel.impl.store.DynamicRecordAllocator;
+import org.neo4j.kernel.impl.store.DynamicStringStore;
+import org.neo4j.kernel.impl.store.PropertyStore;
+import org.neo4j.kernel.impl.store.PropertyType;
+import org.neo4j.kernel.impl.store.StoreFactory;
+import org.neo4j.kernel.impl.store.record.PropertyBlock;
+import org.neo4j.kernel.impl.store.record.PropertyRecord;
+import org.neo4j.kernel.monitoring.Monitors;
+import org.neo4j.logging.LogProvider;
+import org.neo4j.logging.NullLogProvider;
+import org.neo4j.test.EphemeralFileSystemRule;
+import org.neo4j.test.PageCacheRule;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.neo4j.kernel.impl.store.StoreFactory.PROPERTY_STORE_NAME;
+
+@RunWith( Enclosed.class )
+public class StorePropertyCursorTest
+{
+    public static class ErrorTest
+    {
+        private final PropertyStore propertyStore = mock( PropertyStore.class );
+        private final Consumer<StorePropertyCursor> cache = mock( Consumer.class );
+        private final DynamicStringStore stringStore = mock( DynamicStringStore.class );
+        private final DynamicArrayStore arrayStore = mock( DynamicArrayStore.class );
+
+        {
+            when( propertyStore.getStringStore() ).thenReturn( stringStore );
+            when( propertyStore.getArrayStore() ).thenReturn( arrayStore );
+        }
+
+        @Test
+        public void shouldReturnTheCursorToTheCacheOnClose() throws Throwable
+        {
+            // given
+            StorePropertyCursor storePropertyCursor = new StorePropertyCursor( propertyStore, cache );
+
+            // when
+            storePropertyCursor.close();
+
+            // then
+            verify( cache, times( 1 ) ).accept( storePropertyCursor );
+        }
+
+        @Test
+        public void shouldThrowIfTheRecordIsNotInUse() throws Throwable
+        {
+            // given
+            int recordId = 42;
+            PageCursor pageCursor = mock( PageCursor.class );
+            when( propertyStore.newReadCursor( recordId ) ).thenReturn( pageCursor );
+            when( pageCursor.shouldRetry() ).thenReturn( false );
+
+            StorePropertyCursor storePropertyCursor = new StorePropertyCursor( propertyStore, cache );
+
+            try ( Cursor<PropertyItem> cursor = storePropertyCursor.init( recordId ) )
+            {
+                try
```

Comment:
in case if you will have exception in `close()` that will be suppressed. 
here we will have 2 possible exceptions from try block: `NotFoundException` or `AssertionError` that will be raised always.
So root exception will always be exception from try block and in case if we want to check exception from `close()` those will be suppressed, so it looks like we do not need nested try blocks